src/app/server/AppDelegate.h

```diff
@@ -39,4 +40,9 @@ class AppDelegate
      */
     virtual void OnCommissioningWindowOpened() {}
     virtual void OnCommissioningWindowClosed() {}
+    virtual void OnPASEStarted() {}
+    virtual void OnPASEFailed(CHIP_ERROR err) {}
+    virtual void OnPASEComplete() {}
+    virtual void OnCASEFailed(CHIP_ERROR err) {}
+    virtual void OnCASEComplete() {}
```

Comment:
1. These need a bunch of documentation.  Starting with the fact that these are for the _responder_ side of the relevant communications.
2. `OnPASEComplete` is identical to `OnCommissioningSessionStarted`.  Perhaps `OnPASEStarted` should be called `OnCommissioningSessionEstablishmentStarted` and `OnPASEFailed` should be called `OnCommissioningSessionEstablishmentFailed`?
4. There is no "CASE started" notification.  Should there be one?
5. For the CASE notifications, should there be some indication of what the peer node id is?  At least in the "complete" notification (e.g. by passing in the session)?  It's really not clear what the use cases are for these notifications...

Reply:
> these methods aim to expose some notifications related to CASE and PASE on the application side.

OK, but what would applications do with these notifications?

Reply:
1.The above notifications can serve diagnostic purposes.
2.They can be used in the led indicator use case. The led indicators may breathe a different pattern on failure or success through these notifications.
3.They can provide a diagnostic of commissioning failures or success that occur during the commissioning process.

Reply:
>  1.The above notifications can serve diagnostic purposes.

How specifically?  This is important to figure out, because it may affect what data we need to communicate in the notification.

> The led indicators may breathe a different pattern on failure or success through these notifications.

I'd like to understand this better.  Is this for PASE or CASE?

> They can provide a diagnostic of commissioning failures or success that occur during the commissioning process.

This part makes sense, yes.  Is the CASE notification then for the CommissioningComplete bits?

Reply:
The led indicators are for PASE  specifically.
Regarding the CASE session these notifications can serve diagnostic purposes in the following ways:
1. It will help to give a statistical overview. For ex: Out of 10 CASE sessions, how many failed and how many succeeded .
    Suppose 3 failed, then the success ratio is 7/10.
2. Also, for example, the scopedNodeId in the CASEComplete notification will help to track which node and which fabric is being used the most number of times .
These notifications on the application side will not be restricted to logging purposes only.The above use cases can be served as well through these notifications.